### PR TITLE
feature/performance-enhancement

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
## PR Overview
**This PR will address the following Issue/Feature:**
- internal ticket

**This PR will result in the following new package version:**

- v0.7.0: breaking due to materialization changes requiring full refresh

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

> ## 🚨 Breaking Changes 🚨
> - To reduce storage, updated default materialization of staging models to views. 
> >  ⚠️ Running a `--full-refresh` will be required if you have previously run these staging models as tables and get the following error: 
> > ```
> > Trying to create view <model path> but it currently exists as a table. Either drop <model path> manually, or run dbt with `--full-refresh` and dbt will drop it for you.
> > ```
> 
> ## Under the Hood:
> - Included auto-releaser GitHub Actions workflow to automate future releases.
> - Incorporated the new `fivetran_utils.drop_schemas_automation` macro into the end of each Buildkite integration test job.
> - Updated the pull request [templates](https://github.com/fivetran/dbt_jira_source/tree/main/.github).

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] ~dbt run (if incremental models are present) && dbt test~

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned.
- [x] All necessary documentation and version upgrades have been applied.
    
- [x] docs were regenerated (unless this PR does not include any code or yml updates).
- [x] BuildKite integration tests are passing.
- [x] Detailed validation steps have been provided below.

### Detailed Validation
Please share any and all of your validation steps:

- These were only default materialization setting changes and under the hood changes.

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

File: integration_tests/dbt_project.yml [hunk 1]
```diff
@@ -29,6 +29,10 @@ vars:
     jira_user_identifier: "user"
     jira_version_identifier: "version"
 
+models:
+  jira_source:
+    +schema: "{{ 'sqlw_tests' if target.name == 'databricks-sql' else 'jira' }}"
```
Comment (fivetran-joemarkiewicz) [inline integration_tests/dbt_project.yml:34]: Reminder to update the schema name to be unique to Jira for the SQL Warehouse tests

File: integration_tests/requirements.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
-dbt-spark[PyHive]>=1.3.0,<2.0.0
-dbt-databricks>=1.3.0,<2.0.0
\ No newline at end of file
+dbt-snowflake>=1.7.0,<2.0.0
+dbt-bigquery>=1.7.0,<2.0.0
+dbt-redshift>=1.7.0,<2.0.0
+dbt-postgres>=1.7.0,<2.0.0
+dbt-spark>=1.7.0,<2.0.0
+dbt-spark[PyHive]>=1.7.0,<2.0.0
+dbt-databricks>=1.7.0,<2.0.0
```
Comment (fivetran-joemarkiewicz) [inline integration_tests/requirements.txt:7]: Would you be able to elaborate on why this update is needed? If this is needed across the board, then we should probably change the compatible dbt version for the package.

File: macros/is_databricks_sql_warehouse.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{% macro is_databricks_sql_warehouse(target) %}
+    {% if target.type in ('databricks') %}
+        {% set re = modules.re %}
+        {% set path_match = target.http_path %}
+        {% set regex_pattern = "/sql/.+/warehouses/" %}
```
Comment (fivetran-joemarkiewicz) [inline macros/is_databricks_sql_warehouse.sql:5]:
```suggestion
        {% set regex_pattern = "sql/.+/warehouses/" %}
```

**Review by fivetran-catfritz — COMMENTED** [review]
Thanks @fivetran-joemarkiewicz! Updated based on your comments, and this is ready for re-review.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM

**Pull request merged**